dataclasses_json/core.py

```diff
@@ -304,11 +316,14 @@ def _decode_generic(type_, value, infer_missing):
 
         # get the constructor if using corresponding generic type in `typing`
         # otherwise fallback on constructing using type_ itself
-        materialize_type = type_
         try:
             materialize_type = _get_type_cons(type_)
         except (TypeError, AttributeError):
-            pass
+            materialize_type = type_
```

Comment:
Can we wrap this into a method? Would be cleaner and easier to read

Reply:
good point! I have moved the logic to resolve the collection type to decode to in one method now.

Reply:
Is it possible to release/publish the change as well?

Reply:
yes I'll do it on the weekend